### ISSUE TITLE
More cudnn ops

### ISSUE THREAD

**Issue by LaurentMazare** [coreylowman/cudarc#178]
Thanks for this amazing crate, it's been instrumental to candle. We've recently added a feature to use the cudnn conv2d which sped things up a lot compared to our handcrafted kernel, and would like to have cudnn support for more ops. Mostly interested in:
- Conv2d backprop.
- Conv1d forward + backward.
- Maybe flash-attention/softmax/...
Are there any plans to add these to the cudnn safe api? If not would you be ok with people making PR to add it?

**Comment by coreylowman** [comment]
You can already do conv2d backprop with the existing methods (See https://github.com/coreylowman/dfdx/blob/main/src/tensor_ops/conv2d/cudnn_kernel.rs#L91).

Conv1d - I'm not sure this exists in cudnn? Not sure about flash attention existence either (or at least it didn't when I last checked cudnn)

But yes open to any contributions here!

**Comment by LaurentMazare** [comment]
Ah great that the conv2d backward step is already there, we'll add it to candle.

For the conv1d, there is some support for Nd convolution I think, e.g. [in the cudarc ffi](https://github.com/coreylowman/cudarc/blob/4d956e2717b2c71714afb16598514f95a7ee994e/src/cudnn/sys.rs#L3230) so hopefully having a safe api around this would enable 1d convolutions.

For flash attention, I meant [this fused flash attn fprop](https://docs.nvidia.com/deeplearning/cudnn/developer-guide/index.html#flash-fused-multi-head-att-fprop), though I've actually never used it.

**Comment by kckeiks** [comment]
Pooling would be great to have as well. It seems like it could have a similar api as conv.

**Comment by kckeiks** [comment]
@coreylowman would you take a PR for adding nd pooling?

**Comment by coreylowman** [comment]
@kckeiks of course, any and all prs welcome
